compiler/src/dmd/dsymbolsem.d

```diff
@@ -7482,8 +7482,32 @@ private extern(C++) class NewScopeVisitor : Visitor
         }
         sc = sc2;
     }
-}
 
+    override void visit(ClassDeclaration cld)
+    {
+        auto sc2 = cld.newScope(sc);
```

Comment:
I don't think you have translated `super.newScope` here properly. Should be `(cast(AggregateDeclaration)cld).newScope(sc);`